app/src/main/java/com/asfoundation/wallet/ui/balance/BalanceFragmentPresenter.kt

```diff
@@ -31,32 +33,63 @@ class BalanceFragmentPresenter(private val view: BalanceFragmentView,
     handleTokenDetailsClick()
     handleCopyClick()
     handleQrCodeClick()
+    handleBackupClick()
     handleBackPress()
+    handleSetupTooltip()
+    handleTooltipBackupClick()
+    handleTooltipLaterClick()
+  }
+
+  private fun handleTooltipLaterClick() {
+    disposables.add(view.getTooltipDismissClick()
+        .doOnNext {
+          view.dismissTooltip()
+          balanceInteract.saveSeenBackupTooltip()
+        }
+        .subscribe())
+  }
 
+  private fun handleTooltipBackupClick() {
+    disposables.add(view.getTooltipBackupButton()
+        .flatMapSingle { balanceInteract.requestActiveWalletAddress() }
+        .observeOn(viewScheduler)
+        .doOnNext {
+          balanceInteract.saveSeenBackupTooltip()
+          activityView?.navigateToBackupView(it)
+          view.dismissTooltip()
+        }
+        .subscribe({}, { it.printStackTrace() }))
+  }
+
+  private fun handleSetupTooltip() {
+    disposables.add(Single.just(balanceInteract.hasSeenBackupTooltip())
+        .subscribeOn(networkScheduler)
+        .observeOn(viewScheduler)
```

Comment:
observeOn can be only for doOnSuccess and errors right? no need to filter it in the main thread